Functions/xASL_bids_parms2BIDS.m

```diff
@@ -327,8 +321,53 @@
 	end
 end
 
-% Move fields to correct position in x structure
-outParms = xASL_io_CheckDeprecatedFieldsX(outParms);
+% If we convert to xASL format, we have to move fields to correct position in the x structure
+if ~bOutBids
+    outParms = xASL_io_CheckDeprecatedFieldsX(outParms);
+end
 
 
 end
+
+
+% Flatten the x structure to a single level
+function inXasl = xASL_wrp_Quantify_FlattenX(inXasl)
+
+    % Flatten inXasl structure
+    
+    % Move x.Q, x.settings, x.dataset, x.modules, x.modules.structural,
+    % x.modules.asl, and x.modules.population parameters to main level of x
+    
+    firstLevelStructs = {'Q', 'settings', 'dataset', 'modules'};
+    secondLevelStructs = {'structural', 'asl', 'population'};
+    
+    % Iterate over first level structs
+    for iStruct = 1:numel(firstLevelStructs)
+        if isfield(inXasl,firstLevelStructs{iStruct})
+            FieldsQ = fields(inXasl.(firstLevelStructs{iStruct}));
+            for iField=1:length(FieldsQ)
+                inXasl.(FieldsQ{iField}) = inXasl.(firstLevelStructs{iStruct}).(FieldsQ{iField});
+            end
+            inXasl = rmfield(inXasl,firstLevelStructs{iStruct});
+        end
+    end
+    
+    % Iterate over second level structs
```

Comment:
Above is an error! The first-level struct `modules` will be checked for Q-Fields and deleted at the end of the previous cycle. There will never be a chance that this reaches this point of code with anything left in x.modules...

So modules have to be excluded from the first level stuff - or not deleted...

Reply:
147a173